### PR TITLE
Improve Dropout guide

### PR DESCRIPTION
**Pull request by cgarciae**
# What does this PR do?

WIP

### CODE REVIEW

File: docs/contributing.md [hunk 1]
```diff
@@ -129,6 +129,16 @@ To contribute code to Flax on GitHub, follow these steps:
...
 
+## Contribution documentation
+To run the documentation locally, you need to install the requirements in `docs/requirements.txt` as noted above.
+Then, you can run the following command from the top of the repository:
+
+```bash
+sphinx-build -M html docs docs/_build -T
+```
+
+This will build the documentation and place it in `docs/_build/html`. You can then open `docs/_build/html/index.html` in your browser to view the documentation or
```
Comment (8bitmp3) [inline docs/contributing.md:140]: The docs are built in that folder by default. `make html` should work for local builds. It's shorter.

File: docs/README.md [hunk 2]
```diff
@@ -51,10 +51,11 @@ use the following notation:
 ```bash
 # Example code::
```
Comment (ArjunSharda) [inline docs/README.md:52]: Why is there a extra : in "Example Code"? Just curious.

**Review by ArjunSharda — APPROVED** [review]

File: docs/contributing.md (hunk 1 above)
Comment (cgarciae) [reply docs/contributing.md]: I couldn't get `make` to work, also CI doesn't use `make`. Lets include this for now but feel free to send a patch.

File: docs/guides/dropout.rst [hunk 3]
```diff
@@ -124,19 +77,22 @@ In short, :meth:`flax.linen.Module.make_rng` *guarantees full reproducibility*.
       x = nn.Dropout(rate=0.5, deterministic=not training)(x) #!
       return x
 
+**Note**: In other machine learning frameworks, the difference in runtime behavior
```
Comment (8bitmp3) [inline docs/guides/dropout.rst:80]: Please keep the original text, including "TensorFlow (Keras)".

File: docs/guides/dropout.rst [hunk 4]
```diff
@@ -18,84 +16,39 @@ Flax ``Dropout``.
...
-(PRNG) state. Flax uses JAX's (splittable) PRNG keys, which have a number of
-desirable properties for neutral networks. To learn more, refer to the
-`Pseudorandom numbers in JAX tutorial <https://jax.readthedocs.io/en/latest/jax-101/05-random-numbers.html>`__.
-
-**Note:** Recall that JAX has an explicit way of giving you PRNG keys:
-you can fork the main PRNG state (such as ``key = jax.random.PRNGKey(seed=0)``)
-into multiple new PRNG keys with ``key, subkey = jax.random.split(key)``. You
-can refresh your memory in
-`🔪 JAX - The Sharp Bits 🔪 Randomness and PRNG keys <https://jax.readthedocs.io/en/latest/notebooks/Common_Gotchas_in_JAX.html#jax-prng>`__.
+Since dropout uses random masks, it requires a pseudorandom number generator
```
Comment (8bitmp3) [inline docs/guides/dropout.rst:22]: The new explanation raises more questions for less experienced users. Let's keep original text. 

Example: "dropout uses random masks".

Assume not every researcher is proficient in randomness in deep learning, basically.

File: docs/guides/dropout.rst [hunk 5]
```diff
@@ -18,84 +16,39 @@ Flax ``Dropout``.
...
+(PRNG). In JAX PRNG state is explicit and passed around to every
```
Comment (8bitmp3) [inline docs/guides/dropout.rst:23]: Keep the original PRNG explanations until further PRs, as discussed.

File: docs/guides/dropout.rst [hunk 6]
```diff
@@ -18,84 +16,39 @@ Flax ``Dropout``.
...
-
-**Note:** Flax provides an implicit way of handling PRNG key streams via Flax
-:meth:`flax.linen.Module`'s :meth:`flax.linen.Module.make_rng` method.
-This allows you to split off a fresh PRNG key inside Flax Modules (or their
-sub-Modules) from the PRNG stream. The ``make_rng`` method guarantees to provide a
-unique key each time you call it. Internally, :meth:`flax.linen.Dropout` makes
-use of :meth:`flax.linen.Module.make_rng` to create a key for dropout. You can
-check out the
-`source code <https://github.com/google/flax/blob/5714e57a0dc8146eb58a7a06ed768ed3a17672f9/flax/linen/stochastic.py#L72>`__.
-In short, :meth:`flax.linen.Module.make_rng` *guarantees full reproducibility*.
```
Comment (8bitmp3) [inline docs/guides/dropout.rst:98]: Keep until a further separate PR, as discussed.

File: docs/guides/dropout.rst [hunk 7]
```diff
@@ -18,84 +16,39 @@ Flax ``Dropout``.
...
-Split the PRNG key
+PRNGs in JAX
 ******************
 
-Since dropout is a random operation, it requires a pseudorandom number generator
-(PRNG) state. Flax uses JAX's (splittable) PRNG keys, which have a number of
-desirable properties for neutral networks. To learn more, refer to the
-`Pseudorandom numbers in JAX tutorial <https://jax.readthedocs.io/en/latest/jax-101/05-random-numbers.html>`__.
-
-**Note:** Recall that JAX has an explicit way of giving you PRNG keys:
```
Comment (8bitmp3) [inline docs/guides/dropout.rst:29]: Keep until a further separate PR, as discussed

File: docs/README.md (hunk 2 above)
Comment (cgarciae) [reply docs/README.md]: Not sure. However, this is not part of the PR 😅

File: docs/guides/dropout.rst (hunk 5 above)
Comment (cgarciae) [reply docs/guides/dropout.rst:23]: avoided mentioning "masks"
